File: lessons/01-introduction/errors_demo.rs
fn main() {
    //# Syntax error
    // println("hello world");

    // let array = ["hello", "new", "world!"];

    //# Array out-of-bounds with a statically known index
    // println!("{}", array[3]);

    //# Array out-of-bounds with a dynamically computed index
    // for i in 0..=array.len() {
    //     println!("{}", array[i]);
    // }

    //# An unsuccessful attempt at emulating C++'s ability to read the memory we're not supposed to access
    // let format = "a very innocent hello {}";
    // println!(format);

    //# Division by zero
    // let joy_division = 0/0;

    // let joy = 0;
    // let joy_division = 0/joy;

    // let joy = if false {1} else {0};
    // let joy_division = 0/joy;

    // println!("{}", joy_division);
}

File: lessons/old/2021L/01-introduction/errors_demo.c
#include <stdio.h>

#define LENGTH 3

int main() {
    //# Syntax error
    // printf("hello world";

    //char* array[LENGTH] = {"hello", "ancient", "world!"};

    //# Array out-of-bounds with a statically known index
    // printf("%s\n", array[LENGTH]);

    //# Array out-of-bounds with a dynamically computed index
    // for (int i = 0; i <= LENGTH; i++) {
    //     printf("%s\n", array[i]);
    // }

    //# Doing God knows what
    // char* format = "a very innocent hello %s";
    // printf(format);

    //# Division by zero
    // int joy_division = 1/0;

    // int joy = 0;
    // int joy_division = 1/joy;

    // int joy = 0 ? 1 : 0;
    // int joy_division = 1/joy;

    // printf("joy division equals %d", joy_division);

    return 0;
}
